src/os/bluestore/BlueStore.cc

```diff
@@ -3563,12 +3563,11 @@ int BlueStore::_set_cache_sizes()
   double cache_kv_max_ratio = 0;
 
   // if cache_kv_max is negative, disable it
-  if (cache_kv_max > 0) {
-    if (cache_size > 0)
-      cache_kv_max_ratio = (double) cache_kv_max / (double) cache_size;
+  if (cache_size > 0 && cache_kv_max > 0) {
```

Comment:
The original idea was to not apply the max when cache_kv_max is negative.  You are right about the zero case, it should still be applied!